articles/rock-paper-scissor-online/index.md

```diff
@@ -0,0 +1,260 @@
+
+- ```public/index.html``` : A Home Page for the app with all the required UI Components. 
+- ```public/style.css``` : Styles for the entire application.
+- ```public/game.js``` : Client-Side JavaScript file to handle events.
+- ```app.js``` : Server Side NodeJS file for setting up the server.
+
+Note that the app is not functional yet, we need to write the necessary logic to make the app functional. Also, feel free to tweak the boiler code to match your style.
+
+There are two important files to focus on. One is  ```app.js``` that handles server-side logic. And the other is ```public/game.js``` that handles client-side logic.
+
+### Game Flow 
+
+![image](/engineering-education/nginx-reverse-proxy/rps.png)
+
+This is a Sequence diagram that shows the timeline of events. It's important to understand the flow before we start coding. So make sure you understand the diagram completely.
+
+Read through the below points if you are not familiar with Sequence Diagrams.
+
+- Time flows downward.
+- Arrows represent events. The start of Arrow denotes the Emitter, end of arrow denotes the listener. For example, Player1 emits the `Create Game` Event, Socket Server listens to this event.
+- A rectangular box denotes waiting/processing time.
+
+Player1 emits the create Game Event. The server acknowledges and responds back with a room ID. Once player2 emits join Game Event with the same room ID. The server detects that both players have joined the room and shares the other player's info with each other. Once both of the players have their choice, the server processes the result and sends back to both players.
+
+Now that the flow is clear, let's jump right into coding.
+
+### Create Game
+
+Let us start by writing logic for the Create Game Event in the `public/game.js` file.
+
+```
+//Create Game Event Emitter
+$(".createBtn").click(function(){
+    firstPlayer=true;
+    const playerName=$("input[name=p1name").val();
+    socket.emit('createGame',{name:playerName});
+})
+```
+
+The above code listens for a click event on the Create Button. Once that event is triggered, the client grabs player1's name and emits a socket event named `createGame` . The variable `firstPlayer` identifies the player who started the game.  The use of this variable will be explained later.
+
+Now we write server-side code on `app.js` so that the server listens to this event.
+
+```
+io.on("connection",(socket)=>{
+    console.log("connection established");
+
+    //Create Game Listener
+    socket.on("createGame",(data)=>{
+        const roomID=randomstring.generate({length: 4});       
+        socket.join(roomID);        
+        players[roomID]=data.name;
+        socket.emit("newGame",{roomID:roomID});
+    })
+
+})
+```
+
+All event listeners/emitters would go inside the `io.on` block as mentioned above. `connection` is the default event listener provided by Socket.io. A `connection` event is emitted under the hood everytime a connection is established.  
```

Comment:
Remove would

Socket.io and a connect event **- join sentences**

**every time**